### PR TITLE
[Algorithms] Merge sort

### PR DESCRIPTION
**Pull request by Tismoney**

### CODE REVIEW

File: sorting/merge_sort/merge_sort.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        >>> arr = [1, 9, 2, 8, 3, 7, 4, 6, 5]
+        >>> merge_sort(arr)
+        [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    """
+    if len(arr) > 1:
+        m = len(arr) // 2
+        left = merge_sort(arr[:m])
+        right = merge_sort(arr[m:])
+        arr = merge(left, right)    
+    return arr
```
Comment (gkorepanov) [inline sorting/merge_sort/merge_sort.py]: newline at end of file is commonly accepted practice

File: sorting/merge_sort/merge_sort.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+def merge(left, right):
+    """
+    Merge two sorted array into one
+    Parameters
+    ----------
+        left:   list
```
Comment (gkorepanov) [inline sorting/merge_sort/merge_sort.py:8]: one option is to use Python 3's type hints:
```python
from typing import List
...
def merge(left: List, right: List):
    ...
```
It's clear and fully supported by modern IDE's (which helps a lot to write reliable code)
Obviously, you still have to describe non-trivial arguments in docstring, though

File: sorting/merge_sort/merge_sort.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+def merge(left, right):
+    """
+    Merge two sorted array into one
```
Comment (gkorepanov) [inline sorting/merge_sort/merge_sort.py]: Array**s** :)

File: sorting/merge_sort/merge_sort.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    Parameters
+    ----------
+        left:   list
+        right:  list
+    """
+    i = j = 0
+    arr = []
+    while(i < len(left) and  j < len(right)):
```
Comment (gkorepanov) [inline sorting/merge_sort/merge_sort.py]: common practice in python not to use outer brackets, like
```python
while (i < len(left)) and  (j < len(right)):
    ...
```

**Comment by Tismoney** [comment]
Got it. Thanks for comments. Fix it.

**Pull request merged**